windows 10 tutorial

HenryKobin opened the pull request:
Hey guys! I got jekyll running flawlessly on my copy of windows 10 creators update with some pretty basic steps taken. I was curious if I could contribute this guide to the docs. Thanks for your time, let me know if you want me to edit anything / reformat the markdown.

/cc @parkr  @jekyllbot

pathawks commented:
/cc: @jekyll/documentation

tomjohnson1492 commented:
Henry, thanks for writing out this installation doc. I think it would fit better on the [Jekyll on Windows](http://jekyllrb.com/docs/windows/#installation) page rather than in Tutorials. 

Also, how do the rest of the Windows instructions fit in with your Windows 10 section? Do you still have to complete the other Windows-related tasks such as Time Zone Management or Auto-regeneration? Does the Chocolately tutorial still work on Win 10? In short, can you integrate this info more seamlessly into the larger Windows documentation (on that page)?

HenryKobin commented:
Thanks for the response @tomjohnson1492. I agree it would probably fit better on that page. Auto-Regeneration is working fine, although the error message still displays as a warning. I'll have to double check time-zone management, but my files were formatting with the current time/date when I create a new project. 
I am not sure about the Chocolately method, I used Bash so I never had to try. I can certainly double check this. Let me get back to you guys in the next 24 hours with a version of the Jekyll on Windows page. 
Should I create a new pull request when I finish or just continue this one?

DirtyF commented:
@HenryKobin There is no need to create another PR, just continue here. 

/cc @jekyll/windows for feedback

DirtyF commented on docs/_docs/windows.md:
````diff
@@ -8,7 +8,55 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+sudo apt-get install ruby2.3 ruby2.3-dev build-essentials
+```
+
+Or you can install an old version of ruby (1.9.3).
````
I wouldn't mention this, Jekyll requires Ruby > 2.0

pup commented on docs/_docs/windows.md:
````diff
@@ -8,7 +8,56 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+```
+
+
+**And thats it!**
````
`that's`

pup commented on docs/_docs/windows.md:
```diff
@@ -8,7 +8,56 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+
+
+**And thats it!**
+If you cd into the folder, you can make sure time management is working by opening your "_posts"_ folder. You should see a markdown file with the current date listed.
```
cd should be `cd` imo

pup replied:
`"_posts"_` is probably the wrong markup for this? consider using backticks here

pup commented on docs/_docs/windows.md:
```diff
@@ -8,7 +8,56 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+## Installation via Chocolatey
+
 
 A quick way to install Jekyll is to follow the [installation instructions by David Burela](https://davidburela.wordpress.com/2015/11/28/easily-install-jekyll-on-windows-with-3-command-prompt-entries-and-chocolatey/):
```
maybe amend this to say `A quick way to install Jekyll using Chocolatey is [...]`?

pup commented on docs/_docs/windows.md:
````diff
@@ -8,7 +8,56 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+sudo apt-get update
+sudo apt-get install ruby2.3 ruby2.3-dev build-essentials
+```
+Next lets update our Ruby gems:
````
`let's`

pup commented on docs/_docs/windows.md:
````diff
@@ -8,7 +8,56 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+sudo apt-get update -y && sudo apt-get upgrade -y
+```
+
+Now we can install Ruby. First we need to add the repository list from BrightBox.
````
could you provide a link here? i could see folks getting sceptical at just adding a PPA with no real context attached to it

pup commented on docs/_docs/windows.md:
```diff
@@ -8,7 +8,56 @@ Jekyll with the proper tweaks. If you are using Windows 10 Anniversary Update,
+
+*Please note:* You must have [Bash on Ubuntu on Windows](https://msdn.microsoft.com/en-us/commandline/wsl/about?f=255&MSPPError=-2147217396) enabled.
+
+First lets make sure all our packages / repositories are up to date. Open a new Command Prompt instance, and type the following:
```
`let's`

ashmaroli commented:
@HenryKobin can you also test if TimeZone Management is fine during the build process on the subsystem..? I mean, say the `_config.yml` contains `timezone: America/New_York` Does the date and time change accordingly on your `HTML` files in the `_site` folder?
Thanks! :smiley:

HenryKobin commented:
@ashmaroli Timezone management from the config file works just fine. I tested it with America/New_York and it showed up properly via liquid templating.
@pup I corrected those errors, and removed some subheaders from the chocolatey tutorial so that it would be clear that those portions don't apply to the Bash on Ubuntu on Windows situation. I also tested Timezone via the config file, and it showed up correctly in the HTML. Also, auto-regeneration is working.
cc/ @jekyll/windows

DirtyF commented:
Thanks @HenryKobin great addition for Windows users 🐚  

@jekyllbot : merge +docs

DirtyF commented:
@jekyllbot: merge +docs
